tools/ci_build/github/linux/build_cuda_c_api_package.sh

```diff
@@ -1,9 +1,10 @@
 #!/bin/bash
 set -e -x
-docker run --gpus all -e NVIDIA_VISIBLE_DEVICES=all --rm --volume \
+docker run --rm --volume \
 $BUILD_SOURCESDIRECTORY:/onnxruntime_src --volume $BUILD_BINARIESDIRECTORY:/build \
 --volume /data/models:/build/models:ro --volume /data/onnx:/data/onnx:ro -e NIGHTLY_BUILD onnxruntimecuda${CUDA_VERSION_MAJOR}build \
 /usr/bin/python3.9 /onnxruntime_src/tools/ci_build/build.py --enable_lto --build_java --build_nodejs --build_dir /build --config Release \
 --skip_submodule_sync  --parallel --use_binskim_compliant_compile_flags --build_shared_lib --use_cuda --cuda_version=$CUDA_VERSION \
 --cuda_home=/usr/local/cuda-$CUDA_VERSION --cudnn_home=/usr/local/cuda-$CUDA_VERSION \
+--skip_tests \
```

Comment:
@jywu-msft  @yf711 is it OK to skill tests here? There is another test stage later called `Nuget_Test_Linux_GPU` stage and wonder if their test coverages are redundant?

By just skipping this test and using `onnxruntime-Ubuntu2204-AMD-CPU` to build CUDA packages will save whole pipeline build time by 1 hours+, down from 3 - 5 hours to 2 - 3 hours.

Reply:
It is also less expensive and wait time to get an AMD CPU machine than any Nvidia GPU machine.